client/my-sites/patterns/components/category-gallery/style.scss

```diff
@@ -0,0 +1,85 @@
+@import "@automattic/components/src/styles/typography";
+@import "@wordpress/base-styles/breakpoints";
+@import "@wordpress/base-styles/mixins";
+
+.patterns-category-gallery {
+	display: grid;
+	gap: 32px 24px;
+	grid-template-columns: 1fr;
+
+	@include break-medium {
+		grid-template-columns: repeat(calc(var(--column-count) - 1), 1fr);
+	}
+
+	@include break-huge {
+		grid-template-columns: repeat(var(--column-count), 1fr);
+	}
+
+	&.is-regular-patterns {
+		--column-count: 4;
+	}
+
+	&.is-page-patterns {
+		--column-count: 3;
+	}
```

Comment:
The reason I moved this from the component code to the CSS is that we now have several aspects of the `CategoryGallery{Client,Server}` components being controlled by the `patternType` prop. I.e., it makes less sense to treat `columnCount` as something separate. If the design changes now or in the future, we'll need to tweak several aspects anyway.